### PR TITLE
Hit the Mole

### PR DESCRIPTION
**Pull request by tessapower**
Merging this PR will:

- Add a hit area to the mallet
- Create a group for moles to easily identify them as a group
- Detect collisions between the mallet and the mole and call the appropriate functions

### CODE REVIEW

File: scripts/mole/mole.gd [hunk 1]
```diff
@@ -14,15 +14,26 @@ signal mole_died(data: MoleData)
 
 
 var max_health: int:
```
Comment (LongcatOwO) [inline scripts/mole/mole.gd:16]: Does this create a property with public getter but private setter?

File: scripts/mole/mole.gd [hunk 2]
```diff
@@ -14,15 +14,26 @@ signal mole_died(data: MoleData)
...
-	get: return _max_health
+	get: return max_health
```
Comment (LongcatOwO) [inline scripts/mole/mole.gd:17]: Does this create a public getter with private setter?

Comment (tessapower) [reply scripts/mole/mole.gd:17]: I don't believe there is a real difference between private and public variables or functions in Godot - prefixing something with an underscore means that it isn't visible in the public documentation. Hence, by having both a private (`_max_health`) and a public variable (`max_health`) intended to store the same data, you're not actually getting the benefit of restricting access to the private variable, you're just duplicating code and data.

image

These getters and setters are generally used to wrap read/write access and do other things around simply setting the value, e.g. clamping it to a range. Every property is always accessible through dot notation, and this is a bit of syntactic sugar that allows you to go `Obj.get("some_property")`, when `Object.some_property` will achieve the same thing.

Comment (LongcatOwO) [reply scripts/mole/mole.gd:17]: Ah okay

**Pull request merged**